File: platform/ios/demo/Examples/ObjectiveC/CameraFlyToExample.m
@import Mapbox;
#import "CameraFlyToExample.h"

NSString const *MBXExampleCameraFlyTo = @"CameraFlyToExample";

@interface CameraFlyToExample ()
@end

@implementation CameraFlyToExample

- (void)viewDidLoad {
    [super viewDidLoad];
    
    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    
    // Sets Honolulu, Hawaii as the camera's starting point.
    CLLocation *honolulu = [[CLLocation alloc] initWithLatitude:21.3069 longitude:-157.8583];
    [mapView setCenterCoordinate:honolulu.coordinate
                       zoomLevel:14 animated:NO];
    
    mapView.delegate = self;
    [self.view addSubview:mapView];
}

- (void)mapViewDidFinishLoadingMap:(MGLMapView *)mapView {
    
    // Waits for the mapView to finish loading before setting up the camera.
    // Defines the destination camera as Hawaii Island.
    MGLMapCamera *camera = [MGLMapCamera cameraLookingAtCenterCoordinate:CLLocationCoordinate2DMake(19.784213, -155.784605) fromDistance:35000 pitch:70 heading:90];
    
    // Goes from Honolulu to destination camera.
    [mapView flyToCamera:camera withDuration:4.0 peakAltitude:3000 completionHandler:nil];
    // To use default duration and peak altitudes:
    //    [mapView flyToCamera:camera completionHandler:nil];
    // To use default peak altitude:
    //    [mapView flyToCamera:camera withDuration:4 completionHandler:nil];
}
@end

File: platform/ios/demo/Examples/Swift/CameraFlyToExample.swift
import Mapbox

@objc(CameraFlyToExample_Swift)

class CameraFlyToExample_Swift: UIViewController, MGLMapViewDelegate {
    var mapView: MGLMapView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        mapView = MGLMapView(frame: view.bounds)
        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        
        // Sets Honolulu, Hawaii as the camera's starting point.
        let honolulu = CLLocationCoordinate2D(latitude: 21.3069,
                                              longitude: -157.8583)
        mapView.setCenter(honolulu,
                          zoomLevel:14, animated: false)
        
        mapView.delegate = self
        view.addSubview(mapView)
    }
    
    func mapViewDidFinishLoadingMap(_ mapView: MGLMapView) {
        
        // Waits for the mapView to finish loading before setting up the camera.
        // Defines the destination camera as Hawaii Island.
        let camera = MGLMapCamera(lookingAtCenter:
            CLLocationCoordinate2D(latitude: 19.784213, longitude: -155.784605),
                                  fromDistance: 35000, pitch: 70, heading: 90)
        
        // Goes from Honolulu to destination camera.
        mapView.fly(to: camera, withDuration: 4,
                    peakAltitude: 3000, completionHandler: nil)
        // To use default duration and peak altitudes:
        //    mapView.fly(to: camera, completionHandler: nil)
        // To use default peak altitude:
        //    mapView.fly(to: camera, withDuration: 4, completionHandler: nil)
    }
}
